src/test/java/com/puppycrawl/tools/checkstyle/checks/header/HeaderCheckTest.java

```diff
@@ -204,7 +204,9 @@ public void testSetHeaderTwice() {
     @Test
     public void testIoExceptionWhenLoadingHeaderFile() throws Exception {
         final HeaderCheck check = new HeaderCheck();
-        check.setHeaderFile(new URI("test://bad"));
+        final String nonAscii = "\u00e6\u00f8\u00e5";
+        final URI uri = new URI("test://bad" + nonAscii);
+        check.setHeaderFile(uri);
 
```

Comment:
Please create new test method.
We need to make sure all existing tests continue to work as before.